File: 0684/sesión11 - back-end C#/SoftVen/SoftVenPersistance/DAO/ProductoDAO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftVen.Model;

namespace SoftVen.DAO
{
    public interface ProductoDAO
    {
        int Insertar(ProductosDTO producto);

        ProductosDTO ObtenerPorId(int productoId);

        BindingList<ProductosDTO> ListarTodos();

        int Modificar(ProductosDTO producto);

        int Eliminar(ProductosDTO producto);
    }
}

File: 0684/sesión25 - Reportes/Back-End/ServidorRESTFul/SoftVenPersistance/src/main/java/pe/edu/pucp/softven/dao/ProductoDAO.java
package pe.edu.pucp.softven.dao;

import java.util.ArrayList;
import pe.edu.pucp.softven.model.ProductosDTO;

public interface ProductoDAO {

    public Integer insertar(ProductosDTO producto);
    
    public ProductosDTO obtenerPorId(Integer productoId);
    
    public ArrayList<ProductosDTO> listarTodos();
    
    public Integer modificar(ProductosDTO producto);
    
    public Integer eliminar(ProductosDTO producto);
}
